packages/modules/Extension-Ads/src/contentScript/content.js

```diff
@@ -767,14 +767,53 @@ if ( typeof vAPI === 'object' && !vAPI.contentScript ) {
       return content;
     },
     extractAdData(frameId, elem = null) {
-      const { title, company, text } = this.extractTexts(frameId, elem);
+      const { title, text } = this.extractTexts(frameId, elem);
       const content = this.extractContent(frameId, elem);
 
-      return { title, company, text, content };
+      return { title, text, content };
     },
     extractTexts(frameId, element) {
-      // TODO: implement texts extractor
-      return { title: null, company: null, text: null };
+      let title = null;
+      let longestText = '';
+
+      if (!element) {
+        return { title: null, text: null };
+      }
+
+      // Selectors array to identify text elements
+      const selectors = [
+        'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span',
+        'yt-formatted-string', 'body-text', 'title-text'
+      ];
+
+      const elements = Array.from(element.querySelectorAll(selectors)).filter((el) => {
+        return !el.classList.contains('visually-hidden');
+      });
+
+      // Extract title text from the first non-empty element
+      elements.some((el) => {
+        const textContent = el.textContent.trim();
+
+        if (textContent !== '') {
+          title = textContent;
```

Comment:
Don't reassign external variables in array methods that are not intended for it.
In this case just use const title = elements.find( ... ). And remove let title = null; from top.